include/gatekeeper_gk.h

```diff
@@ -99,17 +99,26 @@ struct neighbor_hash_table {
 	/* The tables that store the Ethernet headers. */
 	struct ether_cache *cache_tbl;
 
-	/* TODO Use a spin lock to edit the hash table. */
 	struct rte_hash *hash_table;
 };
 
 /* The gk forward information base (fib). */
 struct gk_fib {
 
+	/* Use a spin lock to edit the FIB table. */
+	rte_spinlock_t lock;
```

Comment:
You should review the presentation you gave about the FIB to remember the discussion we had about the possibility of having lockless readers.